From 72d32c0c12ac014af2fa2275c0aab9eb6eeb81d5 Mon Sep 17 00:00:00 2001
From: Jonathan M Hsieh <jon@cloudera.com>
Date: Tue, 24 Sep 2013 16:47:55 -0700
Subject: [PATCH 01/73] CLOUDERA-BUILD Change build to point to Cloudera versions and repositories.
 - hard code hbase poms to choose hbase-hadoop2-compat module
 - added version string and hardcoded hadoop2 profile for new hbase-thrift module.
 - Change version to 0.96.1.1-cdh5.0.0-SNAPSHOT

---
 hbase-assembly/pom.xml                          |    5 +-
 hbase-assembly/src/main/assembly/components.xml |    6 ++
 hbase-assembly/src/main/assembly/src.xml        |   13 ++++
 hbase-client/pom.xml                            |    6 +-
 hbase-common/pom.xml                            |    6 +-
 hbase-examples/pom.xml                          |    6 +-
 hbase-hadoop-compat/pom.xml                     |    2 +-
 hbase-hadoop1-compat/pom.xml                    |    2 +-
 hbase-hadoop2-compat/pom.xml                    |    2 +-
 hbase-it/pom.xml                                |    8 +-
 hbase-prefix-tree/pom.xml                       |    8 +-
 hbase-protocol/pom.xml                          |    2 +-
 hbase-server/pom.xml                            |   10 ++--
 hbase-shell/pom.xml                             |    8 +-
 hbase-testing-util/pom.xml                      |   12 ++--
 hbase-thrift/pom.xml                            |    8 +-
 pom.xml                                         |   70 ++++++++++++++++++----
 17 files changed, 119 insertions(+), 55 deletions(-)

diff --git a/hbase-assembly/pom.xml b/hbase-assembly/pom.xml
index d615edd..687650d 100644
--- a/hbase-assembly/pom.xml
+++ b/hbase-assembly/pom.xml
@@ -23,7 +23,7 @@
     <parent>
         <artifactId>hbase</artifactId>
         <groupId>org.apache.hbase</groupId>
-        <version>0.96.1.1</version>
+        <version>0.96.1.1-cdh5.0.0-SNAPSHOT</version>
         <relativePath>..</relativePath>
     </parent>
 
@@ -49,6 +49,7 @@
           <tarLongFileMode>gnu</tarLongFileMode>
           <descriptors>
             <descriptor>${assembly.file}</descriptor>
+            <descriptor>src/main/assembly/src.xml</descriptor>
           </descriptors>
         </configuration>
       </plugin>
@@ -94,7 +95,7 @@
     </dependency>
     <dependency>
         <groupId>org.apache.hbase</groupId>
-        <artifactId>${compat.module}</artifactId>
+        <artifactId>hbase-hadoop2-compat</artifactId>
         <version>${project.version}</version>
     </dependency>
     <dependency>
diff --git a/hbase-assembly/src/main/assembly/components.xml b/hbase-assembly/src/main/assembly/components.xml
index ec85bc1..8e44a11 100644
--- a/hbase-assembly/src/main/assembly/components.xml
+++ b/hbase-assembly/src/main/assembly/components.xml
@@ -40,6 +40,12 @@
       </includes>
       <fileMode>0644</fileMode>
     </fileSet>
+    <!-- Include top level cloudera directory -->
+    <fileSet>
+      <directory>cloudera</directory>
+      <fileMode>0644</fileMode>
+      <directoryMode>0755</directoryMode>
+    </fileSet>
     <!-- Include the top level conf directory -->
     <fileSet>
       <directory>${project.basedir}/../conf</directory>
diff --git a/hbase-assembly/src/main/assembly/src.xml b/hbase-assembly/src/main/assembly/src.xml
index b7a21a4..7e93d9b 100644
--- a/hbase-assembly/src/main/assembly/src.xml
+++ b/hbase-assembly/src/main/assembly/src.xml
@@ -65,6 +65,12 @@
               <exclude>.settings/</exclude>
             </excludes>
     </fileSet>
+    <!-- Include top level cloudera directory -->
+    <fileSet>
+      <directory>cloudera</directory>
+      <fileMode>0644</fileMode>
+      <directoryMode>0755</directoryMode>
+    </fileSet>
     <!--Include dev tools-->
     <fileSet>
       <directory>${project.basedir}/../dev-support</directory>
@@ -104,5 +110,12 @@
       </includes>
       <fileMode>0644</fileMode>
     </fileSet>
+    <!-- Include the cloudera directory -->
+    <fileSet>
+      <directory>${project.basedir}/../cloudera</directory>
+      <outputDirectory>cloudera</outputDirectory>
+      <fileMode>0644</fileMode>
+      <directoryMode>0755</directoryMode>
+    </fileSet>
 </fileSets>
 </assembly>
diff --git a/hbase-client/pom.xml b/hbase-client/pom.xml
index 164a7e3..7d85784 100644
--- a/hbase-client/pom.xml
+++ b/hbase-client/pom.xml
@@ -24,7 +24,7 @@
   <parent>
     <artifactId>hbase</artifactId>
     <groupId>org.apache.hbase</groupId>
-    <version>0.96.1.1</version>
+    <version>0.96.1.1-cdh5.0.0-SNAPSHOT</version>
     <relativePath>..</relativePath>
   </parent>
 
@@ -158,7 +158,7 @@
       <activation>
         <property>
             <!--Below formatting for dev-support/generate-hadoopX-poms.sh-->
-            <!--h1--><name>!hadoop.profile</name>
+            <!--h1--><name>hadoop.profile</name><value>1.1</value>
         </property>
       </activation>
       <dependencies>
@@ -178,7 +178,7 @@
       <activation>
         <property>
             <!--Below formatting for dev-support/generate-hadoopX-poms.sh-->
-            <!--h2--><name>hadoop.profile</name><value>2.0</value>
+            <!--h2--><name>!hadoop.profile</name>
         </property>
       </activation>
       <dependencies>
diff --git a/hbase-common/pom.xml b/hbase-common/pom.xml
index b79d453..63e7bb0 100644
--- a/hbase-common/pom.xml
+++ b/hbase-common/pom.xml
@@ -23,7 +23,7 @@
   <parent>
     <artifactId>hbase</artifactId>
     <groupId>org.apache.hbase</groupId>
-    <version>0.96.1.1</version>
+    <version>0.96.1.1-cdh5.0.0-SNAPSHOT</version>
     <relativePath>..</relativePath>
   </parent>
 
@@ -230,7 +230,7 @@
       <activation>
         <property>
             <!--Below formatting for dev-support/generate-hadoopX-poms.sh-->
-            <!--h1--><name>!hadoop.profile</name>
+            <!--h1--><name>hadoop.profile</name><value>1.1</value>
         </property>
       </activation>
       <dependencies>
@@ -273,7 +273,7 @@
       <activation>
         <property>
             <!--Below formatting for dev-support/generate-hadoopX-poms.sh-->
-            <!--h2--><name>hadoop.profile</name><value>2.0</value>
+            <!--h2--><name>!hadoop.profile</name>
         </property>
       </activation>
       <dependencies>
diff --git a/hbase-examples/pom.xml b/hbase-examples/pom.xml
index ac049b4..8565c4c 100644
--- a/hbase-examples/pom.xml
+++ b/hbase-examples/pom.xml
@@ -23,7 +23,7 @@
   <parent>
     <artifactId>hbase</artifactId>
     <groupId>org.apache.hbase</groupId>
-    <version>0.96.1.1</version>
+    <version>0.96.1.1-cdh5.0.0-SNAPSHOT</version>
     <relativePath>..</relativePath>
   </parent>
   <artifactId>hbase-examples</artifactId>
@@ -132,7 +132,7 @@ if we can combine these profiles somehow -->
          <activation>
              <property>
             <!--Below formatting for dev-support/generate-hadoopX-poms.sh-->
-            <!--h1--><name>!hadoop.profile</name>
+            <!--h1--><name>hadoop.profile</name><value>1.1</value>
              </property>
          </activation>
          <dependencies>
@@ -155,7 +155,7 @@ if we can combine these profiles somehow -->
          <activation>
              <property>
             <!--Below formatting for dev-support/generate-hadoopX-poms.sh-->
-            <!--h2--><name>hadoop.profile</name><value>2.0</value>
+            <!--h2--><name>!hadoop.profile</name>
              </property>
          </activation>
          <dependencies>
diff --git a/hbase-hadoop-compat/pom.xml b/hbase-hadoop-compat/pom.xml
index cd8e764..e8b8e15 100644
--- a/hbase-hadoop-compat/pom.xml
+++ b/hbase-hadoop-compat/pom.xml
@@ -23,7 +23,7 @@
     <parent>
         <artifactId>hbase</artifactId>
         <groupId>org.apache.hbase</groupId>
-        <version>0.96.1.1</version>
+        <version>0.96.1.1-cdh5.0.0-SNAPSHOT</version>
         <relativePath>..</relativePath>
     </parent>
 
diff --git a/hbase-hadoop1-compat/pom.xml b/hbase-hadoop1-compat/pom.xml
index c949d0a..4f941ed 100644
--- a/hbase-hadoop1-compat/pom.xml
+++ b/hbase-hadoop1-compat/pom.xml
@@ -21,7 +21,7 @@ limitations under the License.
   <parent>
     <artifactId>hbase</artifactId>
     <groupId>org.apache.hbase</groupId>
-    <version>0.96.1.1</version>
+    <version>0.96.1.1-cdh5.0.0-SNAPSHOT</version>
     <relativePath>..</relativePath>
   </parent>
 
diff --git a/hbase-hadoop2-compat/pom.xml b/hbase-hadoop2-compat/pom.xml
index 8d40f50..4a75dbd 100644
--- a/hbase-hadoop2-compat/pom.xml
+++ b/hbase-hadoop2-compat/pom.xml
@@ -21,7 +21,7 @@ limitations under the License.
   <parent>
     <artifactId>hbase</artifactId>
     <groupId>org.apache.hbase</groupId>
-    <version>0.96.1.1</version>
+    <version>0.96.1.1-cdh5.0.0-SNAPSHOT</version>
     <relativePath>..</relativePath>
   </parent>
 
diff --git a/hbase-it/pom.xml b/hbase-it/pom.xml
index 25daa67..9490e34 100644
--- a/hbase-it/pom.xml
+++ b/hbase-it/pom.xml
@@ -23,7 +23,7 @@
   <parent>
     <artifactId>hbase</artifactId>
     <groupId>org.apache.hbase</groupId>
-    <version>0.96.1.1</version>
+    <version>0.96.1.1-cdh5.0.0-SNAPSHOT</version>
     <relativePath>..</relativePath>
   </parent>
 
@@ -164,7 +164,7 @@
     </dependency>
     <dependency>
       <groupId>org.apache.hbase</groupId>
-      <artifactId>${compat.module}</artifactId>
+      <artifactId>hbase-hadoop2-compat</artifactId>
       <version>${project.version}</version>
     </dependency>
     <dependency>
@@ -219,7 +219,7 @@
       <activation>
         <property>
             <!--Below formatting for dev-support/generate-hadoopX-poms.sh-->
-            <!--h1--><name>!hadoop.profile</name>
+            <!--h1--><name>hadoop.profile</name><value>1.1</value>
         </property>
       </activation>
       <dependencies>
@@ -262,7 +262,7 @@
       <activation>
         <property>
             <!--Below formatting for dev-support/generate-hadoopX-poms.sh-->
-            <!--h2--><name>hadoop.profile</name><value>2.0</value>
+            <!--h2--><name>!hadoop.profile</name>
         </property>
       </activation>
       <dependencies>
diff --git a/hbase-prefix-tree/pom.xml b/hbase-prefix-tree/pom.xml
index 4461327..b221736 100644
--- a/hbase-prefix-tree/pom.xml
+++ b/hbase-prefix-tree/pom.xml
@@ -23,7 +23,7 @@
   <parent>
     <artifactId>hbase</artifactId>
     <groupId>org.apache.hbase</groupId>
-    <version>0.96.1.1</version>
+    <version>0.96.1.1-cdh5.0.0-SNAPSHOT</version>
     <relativePath>..</relativePath>
   </parent>
 
@@ -84,7 +84,7 @@
     </dependency>
     <dependency>
       <groupId>org.apache.hbase</groupId>
-      <artifactId>${compat.module}</artifactId>
+      <artifactId>hbase-hadoop2-compat</artifactId>
       <version>${project.version}</version>
     </dependency>
     <dependency>
@@ -104,7 +104,7 @@
       <activation>
         <property>
             <!--Below formatting for dev-support/generate-hadoopX-poms.sh-->
-            <!--h1--><name>!hadoop.profile</name>
+            <!--h1--><name>hadoop.profile</name><value>1.1</value>
         </property>
       </activation>
       <dependencies>
@@ -138,7 +138,7 @@
       <activation>
         <property>
             <!--Below formatting for dev-support/generate-hadoopX-poms.sh-->
-            <!--h2--><name>hadoop.profile</name><value>2.0</value>
+            <!--h2--><name>!hadoop.profile</name>
         </property>
       </activation>
       <dependencies>
diff --git a/hbase-protocol/pom.xml b/hbase-protocol/pom.xml
index b11e139..2738ddf 100644
--- a/hbase-protocol/pom.xml
+++ b/hbase-protocol/pom.xml
@@ -23,7 +23,7 @@
     <parent>
         <artifactId>hbase</artifactId>
         <groupId>org.apache.hbase</groupId>
-        <version>0.96.1.1</version>
+        <version>0.96.1.1-cdh5.0.0-SNAPSHOT</version>
         <relativePath>..</relativePath>
     </parent>
 
diff --git a/hbase-server/pom.xml b/hbase-server/pom.xml
index 06f51a3..c5c54ba 100644
--- a/hbase-server/pom.xml
+++ b/hbase-server/pom.xml
@@ -23,7 +23,7 @@
   <parent>
     <artifactId>hbase</artifactId>
     <groupId>org.apache.hbase</groupId>
-    <version>0.96.1.1</version>
+    <version>0.96.1.1-cdh5.0.0-SNAPSHOT</version>
     <relativePath>..</relativePath>
   </parent>
   <artifactId>hbase-server</artifactId>
@@ -318,12 +318,12 @@
     </dependency>
     <dependency>
       <groupId>org.apache.hbase</groupId>
-      <artifactId>${compat.module}</artifactId>
+      <artifactId>hbase-hadoop2-compat</artifactId>
       <version>${project.version}</version>
     </dependency>
     <dependency>
       <groupId>org.apache.hbase</groupId>
-      <artifactId>${compat.module}</artifactId>
+      <artifactId>hbase-hadoop2-compat</artifactId>
       <version>${project.version}</version>
       <type>test-jar</type>
       <scope>test</scope>
@@ -522,7 +522,7 @@
       <activation>
         <property>
             <!--Below formatting for dev-support/generate-hadoopX-poms.sh-->
-            <!--h1--><name>!hadoop.profile</name>
+            <!--h1--><name>hadoop.profile</name><value>1.1</value>
         </property>
       </activation>
       <dependencies>
@@ -564,7 +564,7 @@
       <activation>
         <property>
             <!--Below formatting for dev-support/generate-hadoopX-poms.sh-->
-            <!--h2--><name>hadoop.profile</name><value>2.0</value>
+            <!--h2--><name>!hadoop.profile</name>
         </property>
       </activation>
       <dependencies>
diff --git a/hbase-shell/pom.xml b/hbase-shell/pom.xml
index 6059587..ff3d676 100644
--- a/hbase-shell/pom.xml
+++ b/hbase-shell/pom.xml
@@ -23,7 +23,7 @@
   <parent>
     <artifactId>hbase</artifactId>
     <groupId>org.apache.hbase</groupId>
-    <version>0.96.1.1</version>
+    <version>0.96.1.1-cdh5.0.0-SNAPSHOT</version>
     <relativePath>..</relativePath>
   </parent>
   <artifactId>hbase-shell</artifactId>
@@ -181,7 +181,7 @@
     </dependency>
     <dependency>
       <groupId>org.apache.hbase</groupId>
-      <artifactId>${compat.module}</artifactId>
+      <artifactId>hbase-hadoop2-compat</artifactId>
       <version>${project.version}</version>
     </dependency>
     <dependency>
@@ -282,7 +282,7 @@
       <activation>
         <property>
             <!--Below formatting for dev-support/generate-hadoopX-poms.sh-->
-            <!--h1--><name>!hadoop.profile</name>
+            <!--h1--><name>hadoop.profile</name><value>1.1</value>
         </property>
       </activation>
       <dependencies>
@@ -324,7 +324,7 @@
       <activation>
         <property>
             <!--Below formatting for dev-support/generate-hadoopX-poms.sh-->
-            <!--h2--><name>hadoop.profile</name><value>2.0</value>
+            <!--h2--><name>!hadoop.profile</name>
         </property>
       </activation>
       <dependencies>
diff --git a/hbase-testing-util/pom.xml b/hbase-testing-util/pom.xml
index 7ab7567..040e8fd 100644
--- a/hbase-testing-util/pom.xml
+++ b/hbase-testing-util/pom.xml
@@ -23,7 +23,7 @@
     <parent>
         <artifactId>hbase</artifactId>
         <groupId>org.apache.hbase</groupId>
-        <version>0.96.1.1</version>
+        <version>0.96.1.1-cdh5.0.0-SNAPSHOT</version>
         <relativePath>..</relativePath>
     </parent>
     <artifactId>hbase-testing-util</artifactId>
@@ -92,13 +92,13 @@
         </dependency>
         <dependency>
             <groupId>org.apache.hbase</groupId>
-            <artifactId>${compat.module}</artifactId>
+            <artifactId>hbase-hadoop2-compat</artifactId>
             <type>jar</type>
             <scope>compile</scope>
         </dependency>
         <dependency>
             <groupId>org.apache.hbase</groupId>
-            <artifactId>${compat.module}</artifactId>
+            <artifactId>hbase-hadoop2-compat</artifactId>
             <type>test-jar</type>
             <scope>compile</scope>
         </dependency>
@@ -121,7 +121,7 @@
             <activation>
                 <property>
                     <!--Below formatting for dev-support/generate-hadoopX-poms.sh-->
-                    <!--h1--><name>!hadoop.profile</name>
+                    <!--h1--><name>hadoop.profile</name><value>1.1</value>
                 </property>
             </activation>
             <dependencies>
@@ -167,7 +167,7 @@
             <activation>
                 <property>
                     <!--Below formatting for dev-support/generate-hadoopX-poms.sh-->
-                    <!--h2--><name>hadoop.profile</name><value>2.0</value>
+                    <!--h2--><name>!hadoop.profile</name>
                 </property>
             </activation>
             <dependencies>
@@ -253,4 +253,4 @@
             </dependencies>
         </profile>
     </profiles>
-</project>
\ No newline at end of file
+</project>
diff --git a/hbase-thrift/pom.xml b/hbase-thrift/pom.xml
index a67e11f..ba1cfad 100644
--- a/hbase-thrift/pom.xml
+++ b/hbase-thrift/pom.xml
@@ -23,7 +23,7 @@
   <parent>
     <artifactId>hbase</artifactId>
     <groupId>org.apache.hbase</groupId>
-    <version>0.96.1.1</version>
+    <version>0.96.1.1-cdh5.0.0-SNAPSHOT</version>
     <relativePath>..</relativePath>
   </parent>
   <artifactId>hbase-thrift</artifactId>
@@ -185,7 +185,7 @@
     </dependency>
     <dependency>
       <groupId>org.apache.hbase</groupId>
-      <artifactId>${compat.module}</artifactId>
+      <artifactId>hbase-hadoop2-compat</artifactId>
       <version>${project.version}</version>
     </dependency>
     <dependency>
@@ -223,7 +223,7 @@ the same time. -->
       <activation>
         <property>
           <!--Below formatting for dev-support/generate-hadoopX-poms.sh-->
-          <!--h1--><name>!hadoop.profile</name>
+          <!--h1--><name>hadoop.profile</name><value>1.1</value>
         </property>
       </activation>
       <dependencies>
@@ -266,7 +266,7 @@ the same time. -->
       <activation>
         <property>
           <!--Below formatting for dev-support/generate-hadoopX-poms.sh-->
-          <!--h2--><name>hadoop.profile</name><value>2.0</value>
+          <!--h2--><name>!hadoop.profile</name>
         </property>
       </activation>
       <dependencies>
diff --git a/pom.xml b/pom.xml
index d9e94b4..5b74245 100644
--- a/pom.xml
+++ b/pom.xml
@@ -30,16 +30,16 @@
 -->
   <modelVersion>4.0.0</modelVersion>
   <parent>
-    <groupId>org.apache</groupId>
-    <artifactId>apache</artifactId>
-    <version>12</version>
+    <groupId>com.cloudera.cdh</groupId>
+    <artifactId>cdh-root</artifactId>
+    <version>5.0.0-SNAPSHOT</version>
     <relativePath/>
     <!-- no parent resolution -->
   </parent>
   <groupId>org.apache.hbase</groupId>
   <artifactId>hbase</artifactId>
   <packaging>pom</packaging>
-  <version>0.96.1.1</version>
+  <version>0.96.1.1-cdh5.0.0-SNAPSHOT</version>
   <name>HBase</name>
   <description>
     Apache HBase is the Hadoop database. Use it when you need
@@ -356,6 +356,29 @@
   </developers>
   <repositories>
     <repository>
+      <id>cdh.repo</id>
+      <url>https://repository.cloudera.com/artifactory/cloudera-repos</url>
+      <name>Cloudera Repositories</name>
+      <snapshots>
+        <enabled>false</enabled>
+      </snapshots>
+    </repository>
+    <repository>
+      <id>cdh.snapshots.repo</id>
+      <url>https://repository.cloudera.com/artifactory/libs-snapshot-local</url>
+      <name>Cloudera Snapshots Repository</name>
+      <snapshots>
+        <enabled>true</enabled>
+      </snapshots>
+      <releases>
+        <enabled>false</enabled>
+      </releases>
+    </repository>
+    <repository>
+      <id>cloudbees netty</id>
+      <url>http://repository-netty.forge.cloudbees.com/snapshot/</url>
+    </repository>
+    <repository>
       <id>apache release</id>
       <url>https://repository.apache.org/content/repositories/releases/</url>
     </repository>
@@ -482,6 +505,19 @@
             <forkedProcessTimeoutInSeconds>${surefire.timeout}</forkedProcessTimeoutInSeconds>
             <argLine>-enableassertions -Xmx1900m -XX:MaxPermSize=100m -Djava.security.egd=file:/dev/./urandom -Djava.net.preferIPv4Stack=true -Djava.awt.headless=true</argLine>
             <redirectTestOutputToFile>${test.output.tofile}</redirectTestOutputToFile>
+            <includes>
+              <include>${unittest.include}</include>
+            </includes>
+            <excludes>
+              <exclude>${integrationtest.include}</exclude>
+              <exclude>**/*$*</exclude>
+              <exclude>${test.exclude.pattern}</exclude>
+            </excludes>
+            <environmentVariables>
+              <LD_LIBRARY_PATH>${env.LD_LIBRARY_PATH}:${project.build.directory}/nativelib</LD_LIBRARY_PATH>
+              <DYLD_LIBRARY_PATH>${env.DYLD_LIBRARY_PATH}:${project.build.directory}/nativelib</DYLD_LIBRARY_PATH>
+              <MALLOC_ARENA_MAX>4</MALLOC_ARENA_MAX>
+            </environmentVariables>
           </configuration>
           <executions>
             <execution>
@@ -647,6 +683,8 @@
               <exclude>.git/**</exclude>
               <exclude>.svn/**</exclude>
               <exclude>**/.settings/**</exclude>
+              <exclude>**/cloudera/**</exclude>
+              <exclude>**/debian/**</exclude>
             </excludes>
           </configuration>
         </plugin>
@@ -852,7 +890,7 @@
     <buildDate>${maven.build.timestamp}</buildDate>
     <compileSource>1.6</compileSource>
     <!-- Dependencies -->
-    <hadoop-two.version>2.2.0</hadoop-two.version>
+    <hadoop-two.version>${cdh.hadoop.version}</hadoop-two.version>
     <hadoop-one.version>1.1.2</hadoop-one.version>
     <commons-cli.version>1.2</commons-cli.version>
     <commons-codec.version>1.7</commons-codec.version>
@@ -867,8 +905,8 @@
     <guava.version>12.0.1</guava.version>
     <jackson.version>1.8.8</jackson.version>
     <jasper.version>5.5.23</jasper.version>
-    <jaxb-api.version>2.2.2</jaxb-api.version>
-    <jetty.version>6.1.26</jetty.version>
+    <jaxb-api.version>2.1</jaxb-api.version>
+    <jetty.version>6.1.26.cloudera.2</jetty.version>
     <jetty.jspapi.version>6.1.14</jetty.jspapi.version>
     <jersey.version>1.8</jersey.version>
     <jruby.version>1.6.8</jruby.version>
@@ -879,9 +917,9 @@
     <protobuf.version>2.5.0</protobuf.version>
     <stax-api.version>1.0.1</stax-api.version>
     <thrift.version>0.9.0</thrift.version>
-    <zookeeper.version>3.4.5</zookeeper.version>
     <slf4j.version>1.6.4</slf4j.version>
-    <hadoop-snappy.version>0.0.1-SNAPSHOT</hadoop-snappy.version>
+    <zookeeper.version>${cdh.zookeeper.version}</zookeeper.version>
+    <hadoop-snappy.version>${cdh.hadoop-snappy.version}</hadoop-snappy.version>
     <clover.version>2.6.3</clover.version>
     <jamon-runtime.version>2.3.1</jamon-runtime.version>
     <jettison.version>1.3.1</jettison.version>
@@ -921,6 +959,12 @@
     <surefire.secondPartGroups>org.apache.hadoop.hbase.MediumTests</surefire.secondPartGroups>
     <test.output.tofile>true</test.output.tofile>
     <surefire.timeout>900</surefire.timeout>
+    <!-- For flaky tests exclusion -->
+    <test.exclude></test.exclude>
+    <test.exclude.pattern>**/${test.exclude}.java</test.exclude.pattern>
+    <!-- Test inclusion patterns -->
+    <unittest.include>**/Test*.java</unittest.include>
+    <integrationtest.include>**/IntegrationTest*.java</integrationtest.include>
   </properties>
   <!-- Sorted by groups of dependencies then groupId and artifactId -->
   <dependencyManagement>
@@ -965,12 +1009,12 @@
       </dependency>
       <dependency>
         <groupId>org.apache.hbase</groupId>
-        <artifactId>${compat.module}</artifactId>
+        <artifactId>hbase-hadoop2-compat</artifactId>
         <version>${project.version}</version>
       </dependency>
       <dependency>
         <groupId>org.apache.hbase</groupId>
-        <artifactId>${compat.module}</artifactId>
+        <artifactId>hbase-hadoop2-compat</artifactId>
         <version>${project.version}</version>
         <type>test-jar</type>
         <scope>test</scope>
@@ -1458,7 +1502,7 @@
       <activation>
         <property>
             <!--Below formatting for dev-support/generate-hadoopX-poms.sh-->
-            <!--h1--><name>!hadoop.profile</name>
+            <!--h1--><name>hadoop.profile</name><value>1.1</value>
         </property>
       </activation>
       <modules>
@@ -1577,7 +1621,7 @@
       <activation>
         <property>
             <!--Below formatting for dev-support/generate-hadoopX-poms.sh-->
-            <!--h2--><name>hadoop.profile</name><value>2.0</value>
+            <!--h2--><name>!hadoop.profile</name>
         </property>
       </activation>
       <modules>
-- 
1.7.0.4

